modules/openxr/openxr_api_extension.h

```diff
@@ -82,6 +82,23 @@ class OpenXRAPIExtension : public RefCounted {
 	void register_composition_layer_provider(OpenXRExtensionWrapperExtension *p_extension);
 	void unregister_composition_layer_provider(OpenXRExtensionWrapperExtension *p_extension);
 
+	void register_projection_views_extension(OpenXRExtensionWrapperExtension *p_extension);
+	void unregister_projection_views_extension(OpenXRExtensionWrapperExtension *p_extension);
+
+	double get_render_state_z_near();
+	double get_render_state_z_far();
+
+	void set_motion_vector_texture(RID p_render_target);
+	void set_motion_vector_depth_texture(RID p_render_target);
+	void set_motion_vector_target_size(Size2i p_target_size);
+
+	uint64_t openxr_swap_chain_info_create(XrSwapchainCreateFlags p_create_flags, XrSwapchainUsageFlags p_usage_flags, int64_t p_swapchain_format, uint32_t p_width, uint32_t p_height, uint32_t p_sample_count, uint32_t p_array_size);
```

Comment:
Some more API thoughts, specifically about the swapchain create function:

- I think we also need an API to get the list of supported swapchain formats (currently stored in `OpenXRAPI::supported_swapchain_formats` - this should probably get a nice public method to retrieve them).
- We need a way to know which `OpenXRGraphicsExtensionWrapper` is being used so that we know if we're dealing with IDs from OpenGL, Vulkan, Metal, DirectX, etc. Maybe those classes should have a function to get a name string, and then `OpenXRAPI`/`OpenXRAPIExtension` could have a public method to return it to us?

Otherwise, I don't think we have a way to know what values for `p_swapchain_format` would be valid.

Reply:
The whole idea of `OpenXRGraphicsExntesionWrapper` was to hide this externally just like we do in the rest of the rendering engine, you could never write code that takes a future GPU implementation into account so to the outside world, we keep this agnostic. Might not be possible with what we're attempting to do here though, but that all depends on the wider discussion we need to have around whether it even is the right direction for creating a motion vector setup outside of the core that may not align with how motion vectors are implemented within the rendering engine.

Reply:
> you could never write code that takes a future GPU implementation into account so to the outside world, we keep this agnostic.

Yeah, the best you could do is print an error and say "we don't support XXX with YYY backend" and then disable whatever feature needs to do something specific to a particular graphics API.

But in this case, we're not exactly the "outside world", we're somewhere in-between. If we were implementing AppSW within Godot, using `OpenXRSwapChainInfo` and requesting a particular graphics-API-dependent format would be fine. But we're moving into GDExtension, where we are exposing this outside of Godot, but we do want to have the power to do most of the same things we can do within Godot.